src/ScheduledTask/Webhook/CancelOrRefundWebhookHandler.php

```diff
@@ -35,6 +35,8 @@
 
 class CancelOrRefundWebhookHandler implements WebhookHandlerInterface
 {
+    use CancellableWebhookHandlerTrait;
```

Comment:
Is there a reason why we decided to use a trait instead of abstracting this functionality to a helper class?

Reply:
Although I don't really agree that this is cleaner, we can still do this but I think we would need to create another interface something like `CancellableInterface` to be paired with this trait. The interface would require the functions that the trait provides.

This would help easily identify which webhook handlers have the cancellable functionality. What do you think?

Reply:
Just an implementation detail though, but personally I think it's better as a trait than an interface

Reply:
Yeah that's a good point. To go back to my initial point though, for me the biggest issue is that in the trait, the function is basically assuming that there's gonna be an`orderTransactionStateHandler` data member in the class that is using the trait. Whereas if this was in a helper class, this dependency would explicitly be defined in the constructor.

Reply:
Oh yeah that's a good point. I think to fix that I would make the trait's function as dumb as possible and just pass it the handler to use as well. What do you think of that @Morerice ?

Reply:
something like
```php
protected function handleCancelWebhook(
        OrderTransactionEntity $orderTransactionEntity,
        OrderTransactionStateHandler $orderTransactionStateHandler,
        string $state,
        Context $context
    ): void {
        if (PaymentStates::STATE_CANCELLED === $state) {
            $orderTransactionStateHandler->fail($orderTransactionEntity->getId(), $context);
        }
    }
```

Reply:
I think that would be better, but I kinda feel like we're overdoing with abstraction for this case, cause either if we use a helper or a trait, all of these changes will only be to abstract an if condition containing a single function call. Also this if condition will only be useful for the `CancelOrRefund` webhook. What do you think?

Reply:
Resolved offline